sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -613,6 +614,106 @@ function createDefaultRequestPolicyFactories(
   return factories;
 }
 
+export function createPipelineFromOptions(
+  pipelineOptions: InternalPipelineOptions,
+  authPolicyFactory?: RequestPolicyFactory
+) : ServiceClientOptions {
+  let requestPolicyFactories: RequestPolicyFactory[] = [];
+
+  const userAgentInfo: string[] = [];
+  if (pipelineOptions.userAgentOptions && pipelineOptions.userAgentOptions.userAgentPrefix) {
+    userAgentInfo.push(pipelineOptions.userAgentOptions.userAgentPrefix);
+  }
+
+  const defaultUserAgentInfo = getDefaultUserAgentValue();
+  if (userAgentInfo.indexOf(defaultUserAgentInfo) === -1) {
+    userAgentInfo.push(defaultUserAgentInfo);
+  }
+
+  const userAgentOptions = {
+    ...DefaultUserAgentOptions,
+    ...pipelineOptions.userAgentOptions,
+    ...{ userAgentPrefix: userAgentInfo.join(" ") }
```

Comment:
By this time `userAgentInfo` has the actual value we want to set in the header.
Assigning it to `userAgentPrefix` feels wrong.

I would suggest to remove the need for the `userAgentOptions` constant altogether, and pass `userAgentInfo` to the user agent policy directly

Reply:
Yep, you're right, all of that machinery isn't necessary now that we don't deal with the header name any longer.  Fixed this in the latest commit